### PR TITLE
feat: add agent types

### PR DESCRIPTION
**Pull request by vivek3141**
This PR creates an `AgentType` enum for the purposes of being able to create different kinds of agents. Funtionality is checked by a barebones test created in `tests/test_agent_type.py`.

For this modification the `AgentState` is modified to include an `agent_type` field, along with the `CreateAgent` schema that's sent to the server.

### CODE REVIEW

File: letta/client/client.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def agent_exists(self, agent_id: Optional[str] = None, agent_name: Optional[str]
     def create_agent(
         self,
         name: Optional[str] = None,
+        agent_type: Optional[AgentType] = None,
```
Comment (sarahwooders) [inline letta/client/client.py:71]: Could we have this explicitly default to the `AgentType.memgpt_agent`? Just to make it explicit what the defaults are

File: letta/schemas/agent.py [hunk 2]
```diff
@@ -21,6 +22,15 @@ class BaseAgent(LettaBase, validate_assignment=True):
     user_id: Optional[str] = Field(None, description="The user id of the agent.")
 
 
+class AgentType(str, Enum):
+    """
+    Enum to represent the type of agent.
+    """
+
+    base_agent = "base_agent"
```
Comment (sarahwooders) [inline letta/schemas/agent.py:30]: I think this would be better named `memgpt_agent`, since `base_agent` might refer to a common base agent class

**Review by sarahwooders — CHANGES_REQUESTED** [review]

File: letta/client/client.py (hunk 1 above)
Comment (vivek3141) [reply letta/client/client.py:71]: Fixed

File: letta/schemas/agent.py (hunk 2 above)
Comment (vivek3141) [reply letta/schemas/agent.py]: Fixed!

**Review by sarahwooders — APPROVED** [review]
lgtm!

**Pull request merged**